### PR TITLE
Feat/va badge

### LINKED ISSUE #3246: Badge component: improvements and position fixes
**Vuestic-ui version:** 1.6.3

**Steps to reproduce**
!!! please Remove transparent prop from Barge component ---> https://github.com/epicmaxco/vuestic-ui/issues/3262

image
image
image
image

**Other information**
https://www.figma.com/file/PflCWFZC3xz3YCt7uX3MpP/Documentation?node-id=301-1200&t=nNeJcdY6YTDCVeHr-0

### LINKED ISSUE #3254: a11y: va-badge
- [x] step 1: investigate a11y guides for this component;
- [x] step 2: discuss with team;
- [ ] step 3: implement required functionality;
- [ ] step 4: add docs page section about a11y.

### LINKED ISSUE #3262: Remove transparent prop from Badge component
**Is your feature request related to a problem? Please describe.**
Do we need this prop? I can't find any use cases in other frameworks or UI patterns. It seems strange to just add opacity to a component using inline styles
**Discussed: Remove transparent prop from Badge component**

image

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by rustem-nasyrov**
Close #3246
Close #3262
Close #3254

## Description
- Refactoring
- Positioning calculations
- New docs examples

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Review by m0ksem — APPROVED** [review]

File: packages/docs/page-config/ui-elements/badge/index.ts [hunk 1]
```diff
@@ -26,16 +26,16 @@ export default definePageConfig({
       description: "The `dot` property puts the component in a minimalist mode in order to accentuate the base element."
     }),
 
+    block.example("Position", {
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/badge/index.ts:29]: We have position 2 times in docs. Probably this one have to go?

Comment (asvae) [reply packages/docs/page-config/ui-elements/badge/index.ts:29]: Also should be "Placement"

File: packages/docs/page-config/ui-elements/badge/index.ts [hunk 2]
```diff
@@ -26,16 +26,16 @@ export default definePageConfig({
       description: "The `dot` property puts the component in a minimalist mode in order to accentuate the base element."
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/badge/index.ts:26]: Color example looks out of place:

File: packages/ui/src/components/va-badge/hooks/useFloatingPositionStyles.ts [hunk 3]
```diff
@@ -15,9 +17,13 @@ export const useFloatingPositionProps = {
     validator: (position: PlacementWithAlias) => placementsPositionsWithAliases.includes(position),
   },
   offset: {
-    type: [Number, String] as PropType<string | number>,
+    type: [Number, String, Array] as PropType<BadgeOffsetProp>,
```
Comment (asvae) [inline packages/ui/src/components/va-badge/hooks/useFloatingPositionStyles.ts:20]: Offset prop definitely needs more explanation.

Right now there is no examples even.

**Review by asvae — CHANGES_REQUESTED** [review]
Overall pretty good, but there are some critical issues. Let's have one more review after these are fixed.

**Pull request merged**